torch/distributed/_spmd/data_parallel.py

```diff
@@ -155,6 +155,19 @@ def __init__(self, batch_dim: int = 0) -> None:
         # batch dim size is used to track the batch dim size of the input tensor
         self.batch_dim_size = -1
 
+        # reduction ops that is used to detect whether there's a reduction over batch
+        # dimension operation, if there is, we mark the output as sharded instead of
+        # partial placement
+        self.reduction_ops = [
+            aten.binary_cross_entropy.default,
+            aten.binary_cross_entropy_with_logits.default,
+            aten.mean.default,
+            aten.mse_loss.default,
```

Comment:
do we need to test the arguments? If `reduction=None`, it seems there won't be a reduction?

https://pytorch.org/docs/stable/generated/torch.nn.MSELoss.html

Reply:
I see, thinking about it more, I think it should be ok for us, we use `reduction_ops` and whether output_tensor becomes a scalar tensor to jointly determine if there's a reduction on batch dim, so if `reduction=None` it would not be mark as reduction over batch dim